### PR TITLE
Add -D_XOPEN_SOURCE=700 when building musl

### PR DESCRIPTION
**Pull request by sbc100**
This is defined in the upstream Makefile and we should mimik that where
possible.

This has some subtle effects on the build, one of which is not defining
_BSD_SOURCE which in turn means `<strings.h>` is *not* included by
`<string.h>` which allows `__tz.c` to build.

See #9506

### CODE REVIEW

**Comment by kripken** [comment]
Oh, wait - why is this just for libc? In my PR it's for all MuslLibrary libraries, which seems safer?

**Comment by dschuff** [comment]
Is there anything else that we should take from the upstream Makefile?

**Comment by sbc100** [comment]
I moved all these flags to MuslLibrary

**Comment by sbc100** [comment]
The upstream flags are here: https://git.musl-libc.org/cgit/musl/tree/Makefile#n49

Maybe `-std=c99`?  Maybe `-ffreestanding`?  strangely I don't see `-fno-builtin` there...

**Comment by sbc100** [comment]
Oh.. -ffreestandin implies -fno-builtin

**Comment by kripken** [comment]
I'm not sure how moving the `-fno-builtin` will affect other `MuslInternalLibrary` users. We have written `-fno-builtin` specifically where it was needed.

I think #9518 is the safer way to go?

**Comment by sbc100** [comment]
But the whole of musl is designed to built with -fno-builtin (actually -ffreestanding).   

Don't think this is better to have consistent compile flags across the whole of musl?

**Comment by kripken** [comment]
Possibly. However it may be an issue in a library that has both musl and other files, so we'd need to separate things out.

Another issue is that we have picked different optimization levels for different parts of libc, I think intentionally, like the math stuff in `libc_wasm` has `-O2`. That should override the new `-Os` you're adding here, but I'm not sure if other libraries aren't affected.

**Comment by sbc100** [comment]
This change now makes the flags consisten across all the libraries that include musl code.  I think this consistency is helpful.  Unless we see specific regressions around codesize related to Oz vs Os.

**Comment by kripken** [comment]
Ok, lgtm, but I'd recommend running the benchmarks to check for speed or size regressions with this change.

File: tools/system_libs.py [hunk 1]
```diff
@@ -710,7 +712,6 @@ def get_depends(self):
 
 class libc_wasm(MuslInternalLibrary):
   name = 'libc-wasm'
-  cflags = ['-O2', '-fno-builtin']
```
Comment (kripken) [inline tools/system_libs.py]: this part deeply worries me - sin/cos/etc. may not get inlining with `-Os`. Let's keep `-O2` here unless we benchmark carefully and are confident it's ok?

File: tools/system_libs.py [hunk 2]
```diff
@@ -1021,7 +1022,6 @@ class libhtml5(Library):
 class libpthread(AsanInstrumentedLibrary, MuslInternalLibrary, MTLibrary):
   name = 'libpthread'
   depends = ['libc']
-  cflags = ['-O2']
```
Comment (kripken) [inline tools/system_libs.py]: This library may also lose inlining. But maybe it's less worrying here... not sure though. I'd recommend keeping the old flags.

Comment (sbc100) [reply tools/system_libs.py]: This seems kind of arbirary (and I'm not sure it was deliberate).   I don't think inlining is any more desirable for the the code in the pthread subdirectory that any other sub-directory is it?

**Pull request closed without merging**